src/main/scala/com/rbmhtechnology/calliope/EventDeletionSink.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2015 - 2017 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.calliope
+
+import akka.Done
+import akka.stream.scaladsl.Sink
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler}
+import akka.stream.{Attributes, Inlet, SinkShape}
+
+import scala.concurrent.{Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+/**
+  * An EventDeletionSink is an Akka Streams Sink which deletes events through an instance of [[EventDeleter]].
+  * The Sink accepts elements which implement [[Sequenced]] and passes the sequence number of the element to the deletion request sent to the [[EventDeleter]].
+  *
+  * The Sink may be materialized to a [[Future]] to signal completion.
+  */
+object EventDeletionSink {
+  def apply[A: Sequenced](eventWriter: EventDeleter): Sink[A, Future[Done]] =
+    Sink.fromGraph(new EventDeletionSink(eventWriter))
+}
+
+final class EventDeletionSink[In: Sequenced](eventWriter: EventDeleter)(implicit sequenced: Sequenced[In]) extends GraphStageWithMaterializedValue[SinkShape[In], Future[Done]] {
```

Comment:
2) I see, but I think that we run into an conceptual problem with the `flow` method when batching is applied implicitly in the flow - multiple elements are emitted into the flow but the user has no way of determining which element is actually produced by the flow, because of the batching internals.
A `SequencedBatch` would mitigate the problem. 

```
(...)
.batch(20, ev => SequencedBatch(ev)) { (batch, ev) => batch.update(ev) }
.via(EventDeletion.flow[SequencedBatch[A]])
.map(batch => batch.items)
```

Additionally this would give the user the freedom of choice if he wants to use batching, buffering or a timed-approach for deletions.

WDYT?